lib/validateInput.js

```diff
@@ -8,25 +8,60 @@ module.exports = function(inputType) {
 		'lighten', 'lighter', 'color-dodge', 'color-burn', 'hard-light', 'soft-light',
 		'difference', 'exclusion', 'hue', 'saturation', 'color', 'luminosity'];
 
-	switch(inputType) {
+	switch (inputType) {
 		case 'image':
 			// Validate image.position
 			if ((!Array.isArray(this.image.position) || this.image.position.length !== 2) ||
 				xPositionValues.indexOf(this.image.position[0]) === -1 ||
 				yPositionValues.indexOf(this.image.position[1]) === -1
-			) {this.triggerError('image.position')}
+			) { this.triggerError('image.position') }
 			// Validate image.stretchMode
 			if (this.image.stretchMode) {
 				if ((!Array.isArray(this.image.stretchMode) || this.image.stretchMode.length !== 2) ||
 					stretchModeValues.indexOf(this.image.stretchMode[0]) === -1 ||
 					stretchModeValues.indexOf(this.image.stretchMode[1]) === -1
-				) {this.triggerError('image.stretchMode')}
+				) { this.triggerError('image.stretchMode') }
 			}
 			break;
 		case 'blendingMode':
 			if (blendingModeValues.indexOf(this.image.blendingMode) === -1) {
 				this.clear();
 				this.triggerError('blendingMode');
 			}
+			break;
+		case 'direction':
+			if (this.direction === 'custom') {
+				if (!areDefinedAndPixelsOrPercentage([
+					this.customDirection.x0,
+					this.customDirection.x1,
+					this.customDirection.y0,
+					this.customDirection.y1
+				])) {
+					this.triggerError('customDirection');
+				}
+			}
```

Comment:
Can you validate if this is one of the other listed directions, or trigger an error if it's not in the list:
```js
else if (!(this.direction === 'diagonal' || this.direction === 'left-right'  || this.direction === 'top-bottom'  || this.direction === 'radial')) {
	this.triggerError('direction');
}
```

Reply:
Your solution is even better 🙂.